File: 003_longest_substr.py
class Solution:

    def lengthOfLongestSubstring(self, s: str) -> int:
        if (len(set(list(s))) == len(s)):
            return len(s)
        max_len = 0
        for i in range(len(s)):
            curr_substr = set()
            for j in range(i, len(s)):
                prev_len = len(curr_substr)
                curr_substr.add(s[j])
                curr_len = len(curr_substr)

                if (curr_len == prev_len):
                    # Means that it has got a duplicate element, we can just break the loop
                    break
            max_len = max(len(curr_substr), max_len)
        return max_len

File: 003_longest_substr.ts
function lengthOfLongestSubstring(s: string): number {
  let left = 0,
    right = 0;

  let maxLength = 0;

  const set = new Set<string>();

  while (right < s.length) {
    if (!set.has(s[right])) {
      set.add(s[right]);
      right++;
    } else {
      while (true) {
        if (s[left] === s[right]) {
          left++;
          break;
        }
        set.delete(s[left++]);
      }
      set.add(s[right]);
      right++;
    }
    maxLength = Math.max(maxLength, set.size);
  }

  return maxLength;
}
